etsin_finder/qvain_light_resources.py

```diff
@@ -310,21 +310,21 @@ def post(self):
         except ValidationError as err:
             log.warning("Invalid form data: {0}".format(err.messages))
             return err.messages, 400
-        try:
-            metadata_provider_org = session["samlUserdata"][
-                "urn:oid:1.3.6.1.4.1.25178.1.2.9"][0]
-            metadata_provider_user = session["samlUserdata"][
-                "urn:oid:1.3.6.1.4.1.16161.4.0.53"][0]
-        except KeyError as err:
-            log.warning("The Metadata provider is not specified: \n{0}"
-                        .format(err))
-            return {"PermissionError":
-                    "The Metadata provider is not found in login information."}, 401
-        if data["dataCatalog"] == "urn:nbn:fi:att:data-catalog-ida":
+
+        saml_haka_org_id = SAML_ATTRIBUTES.get('haka_org_id')
+        saml_csc_username = SAML_ATTRIBUTES.get('CSC_username')
+        metadata_provider_org = session.get(saml_haka_org_id, [])[0]
```

Comment:
It is supposed to get from the samlUserdata in the session, not directly from the session. Will be fixed in next commit.